### PR TITLE
Add tests for duplicate private methods (early-error)

### PR DESCRIPTION
**Pull request by qmma70**
This PR is for https://github.com/tc39/test262/issues/1343

> Static Semantics: Early Errors
> ClassBody : ClassElementList - It is a Syntax Error if PrivateBoundNames of ClassBody contains any duplicate entries, unless the name is used once for a getter and once for a setter and in no other entries.

We are currently missing test coverage for async functions, generators and async generators. This PR adds coverage for those.

### CODE REVIEW

**Comment by qmma70** [comment]
@jbhoosreddy @caiolima

File: src/class-elements/grammar-privatemeth-duplicate-async-gen.case [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019 Qiming Ma (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: It's a SyntaxError if a class contains a private async generator and a private field with the same name
+info: |
+    Static Semantics: Early Errors
```
Comment (caiolima) [inline src/class-elements/grammar-privatemeth-duplicate-async-gen.case]: This is very nit picking, but we use 2-space indentation.

File: src/class-elements/grammar-privatemeth-duplicate-async.case [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019 Qiming Ma (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: It's a SyntaxError if a class contains a private async function and a private field with the same name
+info: |
+    Static Semantics: Early Errors
```
Comment (caiolima) [inline src/class-elements/grammar-privatemeth-duplicate-async.case]: ditto

File: src/class-elements/grammar-privatemeth-duplicate-gen.case [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019 Qiming Ma (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: It's a SyntaxError if a class contains a private generator and a private field with the same name
+info: |
+    Static Semantics: Early Errors
```
Comment (caiolima) [inline src/class-elements/grammar-privatemeth-duplicate-gen.case]: ditto
